modules/nw-metallb-configure-vrf-bgppeer.adoc

```diff
@@ -0,0 +1,179 @@
+
+The high-level steps to expose a service through a VRF on a BGP peer are the following:
+
+* Define a BGP peer and add a VRF instance.
+* Specify an IP address pool for MetalLB.
+* Configure a BGP route advertisement for MetalLB to advertise a route using the specified IP address pool and the BGP peer associated with the VRF instance.
+* Deploy a service to test the configuration.
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in as a user with `cluster-admin` privileges.
+* You defined a `NodeNetworkConfigurationPolicy` to associate a Virtual Routing and Forwarding (VRF) instance with a network interface. For more information about completing this prerequisite, see the _Additional resources_ section.
+
+.Procedure
+
+. Create `BGPPeer` and `BFDProfile` custom resources (CR).
+
+.. Create a file, such as `frrviavrf.yaml`, with content like the following example:
++
+[source,yaml]
+----
+---
+apiVersion: metallb.io/v1beta2
+kind: BGPPeer
+metadata:
+  name: frrviavrf
+  namespace: metallb-system
+spec:
+  myASN: 100
+  peerASN: 200
+  peerAddress: 192.168.130.1
+  vrf: ens4vrf <1>
+  bfdProfile: testprofile
+---
+apiVersion: metallb.io/v1beta1
+kind: BFDProfile <2>
+metadata:
+  name: testprofile
+  namespace: metallb-system
+----
+<1> Defines the VRF instance to associate with the BGP peer. MetalLB can advertise services and make routing decisions based on the routing information in the VRF.
+<2> Use a BFD profile for the BGP sessions for more rapid failure detection in routes.
+
+.. Apply the configuration for the BGP peer and BFD profile by running the follow command:
++
+[source,terminal]
+----
+$ oc apply -f frrviavrf.yaml
+----
+
+. Create an `IPAddressPool` CR.
+
+.. Create a file, such as `first-pool.yaml`, with content like the following example:
++
+[source,yaml]
+----
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: first-pool
+  namespace: metallb-system
+spec:
+  addresses:
+  - 192.169.10.0/24
+----
+
+.. Apply the configuration for the IP address pool by running the follow command:
++
+[source,terminal]
+----
+$ oc apply -f first-pool.yaml
+----
+
+. Create a `BGPAdvertisement` CR.
+
+.. Create a file, such as `first-adv.yaml`, with content like the following example:
++
+[source,yaml]
+----
+apiVersion: metallb.io/v1beta1
+kind: BGPAdvertisement
+metadata:
+  name: first-adv
+  namespace: metallb-system
+spec:
+  ipAddressPools:
+    - first-pool
+  peers:
+    - frrviavrf <1>
+  nodeSelectors:
+    - matchLabels:
+        egress-service.k8s.ovn.org/test-server1: "" <2>
+----
+<1> In this example, MetalLB advertises a range of IP addresses from the `first-pool` IP address pool to the `frrviavrf` BGP peer. 
+<2> Optional: To use the same return path for the traffic originating from the pod, specify the node to advertise as the next hop for the service. 
```

Comment:
This is not true per se. It's true when using this together with the egress ip feature